### PR TITLE
ISPN-6463 Enable Checkstyle during build

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-6463

### CODE REVIEW

**Comment by slaskawi** [comment]
Rebased

File: parent/pom.xml [hunk 1]
```diff
@@ -1513,7 +1505,19 @@
...
             </configuration>
          </plugin>
-
+         <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-checkstyle-plugin</artifactId>
+            <executions>
+               <execution>
+                  <id>checkstyle</id>
+                  <phase>validate</phase>
```
Comment (gustavonalle) [inline parent/pom.xml]: ```validate``` comes before ```compile```, I think it'd be better to associate with ```verify```

**Comment by gustavonalle** [comment]
@slaskawi Looks good, apart from the lifecycle phase.
Btw, all checks are commented out on ```checkstyle.xml```, so the plugin is not doing anything useful.

File: parent/pom.xml (hunk 1 above)
Comment (slaskawi) [inline parent/pom.xml]: Exactly.. the main though was that if it doesn't fit the rules, it's not worth compiling. What would be the argument for running it in `verify`?

**Comment by slaskawi** [comment]
Yes, most of them are disabled. I'm turning them on gradually

File: parent/pom.xml (hunk 1 above)
Comment (gustavonalle) [inline parent/pom.xml]: My rationale is if the code is not compiling, I'd like to get the compile error first instead of an eventual checkstyle error

Comment (slaskawi) [inline parent/pom.xml]: hmmm yeah it also makes sense... when I think about it again... your approach might be slightly more friendly for those who do not have properly configured IDE... Let's go for it...

**Comment by gustavonalle** [comment]
@slaskawi LGTM, but you added "Changed Required" label?

**Comment by slaskawi** [comment]
whoops missed the lablel button...

**Comment by slaskawi** [comment]
Rebased

**Comment by gustavonalle** [comment]
I believe there's a circularity here: 
the module ```infinispan-checkstyle``` inherits from ```infinispan-parent```, but ```infinispan-parent``` has the checkstyle plugin declared both in the ```pluginManagement``` and in the ```plugin```sections, making ```infinispan-parent``` also a dependant of ```infinispan-checkstyle```.


$ rm -Rf $HOME/.m2/repository/org/infinispan/infinispan-checkstyle
$ mvn -o clean install  -DskipTests=true -s maven-settings.xml
[INFO] Reactor Build Order:
[INFO] 
[INFO] Infinispan BOM
[INFO] Infinispan Common Parent
[INFO] Infinispan Checkstyle Rules
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:checkstyle (checkstyle) on project infinispan-commons: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: Unable to find configuration file at location: checkstyle.xml: Could not find resource 'checkstyle.xml'. -> [Help 1]

To solve the issue, either ```infinispan-parent``` should be prevented to run checkstyle or ```infinispan-checkstyle``` should not inherit from parent

**Comment by danberindei** [comment]
@gustavonalle @slaskawi I'd rather change infinispan-checkstyle to not inherit infinispan-parent.

**Comment by gustavonalle** [comment]
@danberindei @slaskawi the other alternative would be to keep checkstyle only on ```pluginManagement``` and enable it on a per module basis, that should work as well

**Comment by slaskawi** [comment]
ooh I missed that... Running with `-o` is really neat here... Let me think about some alternatives and get back to it shortly. Currently I'm leaning towards to @danberindei's idea.

**Comment by slaskawi** [comment]
I changed checkstyle module's parent from `infinispan-parent` to `infinispan-bom`

**Comment by gustavonalle** [comment]
Looks good, pulling...

**Comment by gustavonalle** [comment]
integrated, thanks @slaskawi!

**Pull request closed without merging**